src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php

```diff
@@ -59,56 +60,85 @@ public function getCustomMetrics($accountId, $webPropertyId)
         return $body['items'];
     }
 
-    public function request($method, $url, $body = null)
+    public function request($method, $url, $body = null, $query = null)
     {
         $this->apiCallsCount++;
+
+        $options = !is_null($body) ? ['json' => $body] : ['query' => $query];
+
         $response = $this->api->request(
             $url,
             $method,
             ['Accept' => 'application/json'],
-            ['json' => $body]
+            $options
         );
 
         return json_decode($response->getBody()->getContents(), true);
     }
 
-    /**
-     * @param $query
-     *
-     * array of arrays
-     *   - viewId - profile / view ID,
-     *   - metrics - array of metrics
-     *   - dimensions - array of dimensions [OPTIONAL]
-     *   - filtersExpression - filter expression [OPTIONAL]
-     *   - segments - segment ID [OPTIONAL]
-     *   - dateRanges - array of Date ranges
-     *   - orderBy - dimension or metric to order by
-     *
-     * @return array
-     * @throws \Keboola\Google\ClientBundle\Exception\RestApiException
-     */
+    public function getReport($query)
```

Comment:
nemelo by se spis tohle jmenovat getBatch a to getBatch jmenovat getReport?

Reply:
Mozno mas pravdu, ja som to bral tak, ze ta funkcia getBatch() robi request na `*:batchGet` aj ked sa ta konstanta vola REPORTS_URL :)

https://github.com/keboola/google-analytics-extractor/pull/40/files/8d4c95e2d1b05c00e5648e76315396e8d2939c19#diff-9577aef28caef4c01677a5d2c81b9fddR17

Este sa nad tym zamyslim ;)